### PR TITLE
DRAFT Compiler FE: Support ResizeBilinear operator

### PR DESCRIPTION
**Pull request by kvochko**
This enables support for ResizeBilinear operator in compiler frontend

For #1476 

ONE-DCO-1.0-Signed-off-by: Andrey Kvochko a.kvochko@samsung.com

### CODE REVIEW

File: compiler/luci/import/include/luci/Import/Nodes/CircleResizeBilinear.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  bool validate(const ValidateArgs &args) const final;
+
+private:
+  CircleNode *build_node(const circle::OperatorT &op, const std::vector<CircleNode *> &inputs,
+                         loco::Graph *graph) const final;
+};
+
+} // namespace luci
+
+#endif // __LUCI_IMPORT_OP_CIRCLE_RESIZE_BILINEAR_H__
```
Comment (seanshpark) [inline compiler/luci/import/include/luci/Import/Nodes/CircleResizeBilinear.h:37]: Needs empty last line here

File: compiler/luci/import/src/Nodes/CircleResizeBilinear.cpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  node->input(inputs[0]);
+  node->size(inputs[1]);
+
+  const auto *options = op.builtin_options.AsResizeBilinearOptions();
+  node->align_corners(options->align_corners);
+
+  return node;
+}
+
+} // namespace luci
```
Comment (seanshpark) [inline compiler/luci/import/src/Nodes/CircleResizeBilinear.cpp:50]: here too

File: compiler/luci/lang/include/luci/IR/Nodes/CircleResizeBilinear.h [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  bool align_corners() const { return _align_corners; }
+  void align_corners(bool value) { _align_corners = value; }
+
+private:
+  bool _align_corners = false;
+};
+
+} // namespace luci
+
+#endif // __LUCI_IR_CIRCLERESIZE_BILINEAR_H__
```
Comment (seanshpark) [inline compiler/luci/lang/include/luci/IR/Nodes/CircleResizeBilinear.h:54]: here too

File: compiler/luci/lang/src/Nodes/CircleResizeBilinear.test.cpp [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+{
+  luci::CircleResizeBilinear resize;
+
+  ASSERT_EQ(luci::CircleDialect::get(), resize.dialect());
+  ASSERT_EQ(luci::CircleOpcode::RESIZE_BILINEAR, resize.opcode());
+
+  ASSERT_EQ(nullptr, resize.input());
+  ASSERT_EQ(nullptr, resize.size());
+  ASSERT_EQ(false, resize.align_corners());
+}
```
Comment (seanshpark) [inline compiler/luci/lang/src/Nodes/CircleResizeBilinear.test.cpp:33]: here too

File: compiler/tflchef/core/src/Op/ResizeBilinear.cpp [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+  options_builder.add_align_corners(options.align_corners());
+
+  return options_builder.Finish().Union();
+}
+
+std::unique_ptr<OpChef> ResizeBilinearChefFactory::create(const tflchef::Operation *operation) const
+{
+  return std::unique_ptr<OpChef>{new ResizeBilinearChef{operation}};
+}
```
Comment (seanshpark) [inline compiler/tflchef/core/src/Op/ResizeBilinear.cpp:41]: here too

File: compiler/tflchef/tflite/src/Op/ResizeBilinear.cpp [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  operation->set_type("ResizeBilinear");
+
+  auto op_options = operation->mutable_resize_bilinear_options();
+
+  op_options->set_align_corners(op_params->align_corners());
+
+  return operation;
+}
+
+} // namespace tflchef
```
Comment (seanshpark) [inline compiler/tflchef/tflite/src/Op/ResizeBilinear.cpp:59]: here too

File: compiler/tflchef/tflite/src/Op/ResizeBilinear.h [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+public:
+  void filler(const tflite::Operator *op, TFliteImport *import,
+              tflchef::ModelRecipe *model_recipe) const override;
+  tflchef::Operation *build(const tflite::Operator *op, TFliteImport *import,
+                            tflchef::ModelRecipe *model_recipe) const override;
+};
+
+} // namespace tflchef
+
+#endif // __TFLITE_OP_RESIZE_BILINEAR_H__
```
Comment (seanshpark) [inline compiler/tflchef/tflite/src/Op/ResizeBilinear.h:39]: here too

File: compiler/tflite2circle/src/BuildBuiltinOptions/ResizeBilinearOptions.cpp [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+{
+  auto tflite_builtin_options = op->builtin_options_as_ResizeBilinearOptions();
+  assert(tflite_builtin_options);
+
+  circle::ResizeBilinearOptionsBuilder builtin_options_builder{fb};
+  builtin_options_builder.add_align_corners(tflite_builtin_options->align_corners());
+  return builtin_options_builder.Finish();
+}
+
+} // namespace tflite2circle
```
Comment (seanshpark) [inline compiler/tflite2circle/src/BuildBuiltinOptions/ResizeBilinearOptions.cpp:35]: here too

File: compiler/tflite2circle/src/BuildBuiltinOptions/ResizeBilinearOptions.h [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace tflite2circle
+{
+
+flatbuffers::Offset<circle::ResizeBilinearOptions>
+build_circle_ResizeBilinearOptions(flatbuffers::FlatBufferBuilder &fb, const tflite::Operator *op);
+
+} // namespace tflite2circle
+
+#endif // __BBO_RESIZE_BILINEAR_OPTIONS_H__
```
Comment (seanshpark) [inline compiler/tflite2circle/src/BuildBuiltinOptions/ResizeBilinearOptions.h:31]: here too

**Comment by seanshpark** [comment]
Looks OK with some trivial format changes; Let's proceed PRs;

**Comment by seanshpark** [comment]
Can we close this draft? or are there any left to land?

**Comment by kvochko** [comment]
It's all done!

**Pull request closed without merging**